File: backend/src/NLC.Application/DTOs/Settings/SettingsDtos.cs
using NLC.Core.Enums;

namespace NLC.Application.DTOs.Settings;

// ── Warehouses ────────────────────────────────────────────────────────────────
public record WarehouseDto(Guid Id, string Name, string Location, bool IsActive);
public record CreateWarehouseRequest(string Name, string Location);
public record UpdateWarehouseRequest(string Name, string Location, bool IsActive);

// ── Job Type Configs ──────────────────────────────────────────────────────────
public record JobTypeConfigDto(
    Guid     Id,
    string   Name,
    string[] Phases,
    bool     VasOptional,
    string?  GrnTriggerPhase,
    string?  ErpPushPhase,
    bool     IsActive
);
public record UpdateJobTypeConfigRequest(
    string[] Phases,
    bool     VasOptional,
    string?  GrnTriggerPhase,
    string?  ErpPushPhase,
    bool     IsActive
);

// ── Users ─────────────────────────────────────────────────────────────────────
public record AppUserDto(
    Guid     Id,
    string   Email,
    string   Name,
    UserRole Role,
    Guid[]   AssignedWarehouseIds,
    bool     IsActive
);
public record CreateUserRequest(
    string   Email,
    string   Password,
    string   Name,
    UserRole Role,
    Guid[]   AssignedWarehouseIds
);
public record UpdateUserRequest(
    string   Name,
    UserRole Role,
    Guid[]   AssignedWarehouseIds,
    bool     IsActive
);
public record ChangePasswordRequest(string CurrentPassword, string NewPassword);

// ── ERP Config ────────────────────────────────────────────────────────────────
public record ErpConfigDto(string ErpApiUrl, int LaborRateAed, bool DemoMode);
public record UpdateErpConfigRequest(string ErpApiUrl, int LaborRateAed, bool DemoMode);

File: backend/src/main/java/com/nlc/web/dto/SettingsDtos.java
package com.nlc.web.dto;

import com.nlc.domain.enums.Enums.UserRole;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import java.util.List;
import java.util.Map;
import java.util.UUID;

public final class SettingsDtos {
    private SettingsDtos() {}

    // Warehouse
    public record WarehouseDto(UUID id, String name, String location, boolean isActive) {}

    public record CreateWarehouseRequest(@NotBlank String name, @NotBlank String location) {}

    public record UpdateWarehouseRequest(@NotBlank String name, @NotBlank String location, boolean isActive) {}

    // Job Type Config
    public record JobTypeConfigDto(
        UUID id, String name, String[] phases,
        boolean vasOptional, String grnTriggerPhase,
        String erpPushPhase, boolean isActive
    ) {}

    public record UpdateJobTypeConfigRequest(
        String[] phases, boolean vasOptional,
        String grnTriggerPhase, String erpPushPhase, boolean isActive
    ) {}

    // Users
    public record AppUserDto(
        UUID id, String email, String name,
        String role, String[] assignedWarehouseIds, boolean isActive
    ) {}

    public record CreateUserRequest(
        @NotBlank @Email String email,
        @NotBlank String name,
        @NotBlank String password,
        @NotNull String role,
        String[] assignedWarehouseIds
    ) {}

    public record UpdateUserRequest(
        @NotBlank String name,
        @NotNull String role,
        String[] assignedWarehouseIds,
        boolean isActive
    ) {}

    public record ChangePasswordRequest(
        @NotBlank String currentPassword,
        @NotBlank String newPassword
    ) {}

    // System Config
    public record SystemConfigDto(Map<String, String> settings) {}
}
